include/raft.h.in

```diff
@@ -1022,6 +1022,153 @@ RAFT_API void raft_seed(struct raft *r, unsigned random);
  * The @update output parameter contains changes that user code should perform
  * after the raft_step() call returns (e.g. new entries that must be persisted,
  * new messages that must be sent, etc.).
+ *
+ * Memory Ownership Model for raft_event Fields
+ * =============================================
+ *
+ * The caller must understand the memory ownership semantics for each event type
+ * to avoid memory leaks or use-after-free bugs. This section documents who owns
+ * the memory for each field in the raft_event union after raft_step() returns.
+ *
+ * RAFT_START Event
+ * ----------------
+ * - event->start.term: Scalar value (no ownership concerns)
+ * - event->start.voted_for: Scalar value (no ownership concerns)
+ * - event->start.metadata: PARTIAL OWNERSHIP TRANSFER. The nested configuration
+ *   field is transferred to raft (moved into r->configuration). The metadata
+ *   structure itself is NOT freed by raft and must be freed by the caller after
+ *   raft_step() returns, regardless of success or error. The caller must NOT
+ *   close the configuration field after the call, as it has been transferred.
+ * - event->start.start_index: Scalar value (no ownership concerns)
+ * - event->start.entries: OWNERSHIP TRANSFERRED to raft. On error, raft_step()
+ *   automatically frees the entries via entryBatchesDestroy(). On success, the
+ *   entries are consumed. The caller must NOT free the entries after calling
+ *   raft_step(), regardless of success or error. The entries must be allocated
+ *   such that all entries in a batch share the same batch pointer, and each
+ *   distinct batch pointer appears only once in the entries array for proper
+ *   cleanup.
```

Comment:
This is correct. However I think the behavior of the error handling part is unintended and just result of the v0 to v1 API refactoring.

I would like to change this to be consistent with what happens with `event->start.metadata`, in the sense that the ownership of the `entries` pointer passed to `raft_step()` stays in the hands of the caller.

Basically the idea is that for all data passed to `RAFT_START` via a pointer (i.e. `start.metadata` and `start.entries`) is passed read-only and the ownership is NOT transferred. This should be true for the pointer itself and for nested pointers too.